### ISSUE TITLE
TPU/Trainium support

### ISSUE THREAD

**Issue by BrownianNotion** [axolotl-ai-cloud/axolotl#3521]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Hey team, 
Creating a separate issue for this [thread](https://github.com/axolotl-ai-cloud/axolotl/issues/3440#issuecomment-4090492530). I was wondering if there were any plans for including TPU/Trainium support in the near future? 

Axolotl is currently on accelerate v1.13.0, which supports both platforms. I have linked relevant documentation below
1. Trainium support was just added in [1.13.0](https://github.com/huggingface/accelerate/releases/tag/v1.13.0).
2. TPU support has existed for longer, but isn't particularly well documented.
https://huggingface.co/docs/accelerate/basic_tutorials/tpu
https://huggingface.co/docs/accelerate/concept_guides/training_tpu

@NanoCode012 mentioned potential issues such as difficulty including the platform in dev/CI. Would be great to hear what the team thinks and if TPU would be feasible!

### ✔️ Solution

Support the same, or a useful subset of, the features available in accelerate for TPU/Trainium training (eg. single node multi-accelerator parallel training). This would be great as these chips have better cost ratios than GPUs, in spite of their lower availability.

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by BrownianNotion** [comment]
It looks like Trainium is still too early, integration with TRL is still WIP https://github.com/huggingface/trl/pull/5125. 
However, once it's available, I imagine it would be a lot easier to support than TPUs, as it is supported in native PyTorch and doesn't require torch xla.

**Comment by BrownianNotion** [comment]
I stand corrected, Trainium should already be supported in transformers/peft/accelerate https://github.com/huggingface/trl/pull/5125#issuecomment-4216040424

**Comment by xikronz** [comment]
hai @ved1beta, doesn't seem like i can do '/assign' for this issue but would love to work on it!! can u add me to the assignees?

**Comment by ved1beta** [comment]
just submit a pr !
